plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -234,23 +234,24 @@ public function test_disable_responsive_image_with_picture_element(): void {
 
 		$img_markup = apply_filters( 'the_content', $image );
 
-		// Apply picture element support.
-		$this->opt_in_to_picture_element();
-
-		$picture_markup = apply_filters( 'the_content', $image );
-
 		$img_processor = new WP_HTML_Tag_Processor( $img_markup );
 		$img_processor->next_tag( array( 'tag_name' => 'IMG' ) );
 
-		$this->assertNull( $img_processor->get_attribute( 'sizes' ), 'Sizes attribute missing in IMG tag.' );
+		$this->assertNull( $img_processor->get_attribute( 'sizes' ), 'Make sure that there is no sizes attribute in IMG tag.' );
+		$this->assertNull( $img_processor->get_attribute( 'srcset' ), 'Make sure that there is no srcset attribute in IMG tag.' );
```

Comment:
Nit: It's ok to have these assertions here, but really what we want the tests to focus on is that when the image doesn't have `srcset` or `sizes` then the image does not get modified by our function that wraps the image markup in  a `picture` element with `source` elements for each MIME type. 

We're not really testing that returning false from the `wp_calculate_image_sizes` filter  results in no responsive image. A simpler way of doing this is skip running the first `apply_filters()` function above (so no dynamic image modifications are made), then pass the image directly to `webp_uploads_wrap_image_in_picture()` with 'the_content' set as context and test that the image hasn't been modified.

Reply:
Thanks. Update the logic per https://github.com/WordPress/performance/issues/1448#issuecomment-2276202488.